pyuvdata/uvdata/mir_parser.py

```diff
@@ -3042,33 +3186,38 @@ def __eq__(self, other, verbose=True, metadata_only=False):
                         )
                     continue
 
+                comp_list = data_comp_dict[item]
+
                 # For the attributes with multiple fields to check, list them
                 # here for convenience.
-                comp_dict = {
-                    "raw_data": ["raw_data", "scale_fac"],
-                    "vis_data": ["vis_data", "vis_flags"],
-                }
                 for key in this_attr:
                     this_item = this_attr[key]
                     othr_item = other_attr[key]
 
-                    item_same = False
-                    # auto_data entries are just ndarrays, which we can compare directly
-                    if item == "auto_data":
-                        if this_item.shape == othr_item.shape:
-                            item_same = np.allclose(this_item, othr_item)
-                    else:
-                        # If cross-correlation data, then there are multiple dict
-                        # entries that we need to compare (defined above).
-                        for subkey in comp_dict[item]:
-                            if subkey == "scale_fac":
-                                item_same = this_item[subkey] == othr_item[subkey]
-                            else:
-                                if this_item[subkey].shape == othr_item[subkey].shape:
-                                    item_same = np.allclose(
-                                        this_item[subkey], othr_item[subkey]
-                                    )
-                    if not item_same:
+                    is_same = True
+                    for subkey in comp_list:
+                        if subkey == "scale_fac":
+                            is_same &= this_item[subkey] == othr_item[subkey]
+                        elif subkey == "flags":
+                            is_same &= np.array_equal(
+                                this_item[subkey], othr_item[subkey]
+                            )
+                        else:
+                            # The atol here is set by the max value in the spectrum
+                            # times 2^-10. That turns out to be _about_ the worst case
+                            # scenario for moving to and from the raw data format, which
+                            # compresses the data down from floats to ints.
+                            try:
+                                is_same &= np.allclose(
```

Comment:
So `ValueError` will get raised if the two inputs of `numpy.allclose` are not of the same shape (or otherwise are not broadcastable) -- but in taking a second look at it, I realized that I was getting a little too cute with the `try/except` statement here (e.g., may erroneously work for shape `(1,)` vs `(N,)`). So I put something in that's more explicit for checking the shape and also works out to be faster (in 19c3b3c2).